### PR TITLE
Add command line option to run without audio

### PR DESCRIPTION
**Pull request by torarnv**
The default behavior is to run with audio if we're showing the Tart UI window. It can be disabled with --no-audio.

When running without a UI Window, audio is disabled, but can be enabled via --audio.

### CODE REVIEW

File: Sources/tart/Commands/Run.swift [hunk 1]
```diff
@@ -36,6 +36,16 @@ struct Run: AsyncParsableCommand {
...
 
+  @Flag(help: ArgumentHelp(
+    "Pass audio through to host.",
+    discussion: "This is the default if opening a UI window."))
+  var audio: Bool = false
+
+  @Flag(help: ArgumentHelp(
+    "Don't pass audio through to host.",
+    discussion: "Disables audio pass through, even when opening a UI window."))
+  var noAudio: Bool = false
```
Comment (fkorotkov) [inline Sources/tart/Commands/Run.swift:47]: There is no need in `noAudio`. SwiftArguments will handle `--no-audio` if you have `--audio`. You can leave only `audio` variable and default it to `true` for backward compatibility.

File: Sources/tart/Commands/Run.swift [hunk 2]
```diff
@@ -204,13 +218,18 @@ struct Run: AsyncParsableCommand {
       serialPorts.append(createSerialPortConfiguration(tty_read!, tty_write!))
     }
 
+    let useVNC = vnc || vncExperimental
+    let useBuiltInGraphics = (useVNC && graphics) || !noGraphics
+    let useAudio = (useBuiltInGraphics && !noAudio) || audio;
```
Comment (fkorotkov) [inline Sources/tart/Commands/Run.swift:223]: `useVNC && graphics` seems a bit off. Can we just start passing `audio` down to `craftConfiguration` without changing these variables and logic.

**Review by fkorotkov — COMMENTED** [review]
Thank you for the improvement! Great idea to allow control audio in the VM. IMO overthinking with separate `--no-audio` and `--audio` with custom logic for VNC/nonVNC is too complex and magical. What do you think of a single `audio` flag that will allow to directly control audio without customization for window/no window?

File: Sources/tart/Commands/Run.swift (hunk 1 above)
Comment (torarnv) [reply Sources/tart/Commands/Run.swift:47]: Ah, great! I modeled the two variables after `graphics` and `noGraphics`.

**Comment by torarnv** [comment]
> Thank you for the improvement! Great idea to allow control audio in the VM. IMO overthinking with separate --no-audio and --audio with custom logic for VNC/nonVNC is too complex and magical. What do you think of a single audio flag that will allow to directly control audio without customization for window/no window?

I think the expectation when running with `--no-graphics` is that Tart is running as a service, where the host is not affected. Having random sounds come out of the host sound device because some audio based CI test in the guest is running seems strange with that expectation 😄 

If disabling audio in no-graphics mode is too magical, perhaps we should introduce a `--background` or `--daemon` or `--detach` flag, that disables both graphics and audio, and also backgrounds Tart instead of running in the foreground? It's something I've missed, similar to `docker start`, where I can kick off Tart in the background without fiddling with stuff like `nohup tart run something >/dev/null 2>&1 &`.

File: Sources/tart/Commands/Run.swift (hunk 1 above)
Comment (fkorotkov) [reply Sources/tart/Commands/Run.swift:47]: Oh. Maybe I confused SwiftArguments features with what we use for Go. Regardless, I think we can leave just one variable either `audio = true` or `noAudio = false`. And pass it down to `craftConfiguration`.

**Comment by fkorotkov** [comment]
I did a little digging and `--graphics` was introduced in #248 to support having both VNC and UI for debugging Packer plugin in https://github.com/cirruslabs/packer-plugin-tart/pull/21 This way Packer plugin runs something like `tart run sonoma-base --graphics --vnc-experimental` to have both VNC for the plugin to send commands to and UI window for debugging by a person.

In case of audio still seems like a single `--no-audio` mimicking `--no-graphics` is sufficient enough.

I think in most of the cases when `--no-graphics` is used it's used on a host somewhere in a datacenter where either sounds on the host is already disabled or there are no people to bother. 😅 Introducing yet another flag like `--headless` is an option but might be too much for just passing `--no-graphics --no-audio` by an integrator.

**Comment by torarnv** [comment]
> I did a little digging and `--graphics` was introduced in #248 to support having both VNC and UI for debugging Packer plugin in [cirruslabs/packer-plugin-tart#21](https://github.com/cirruslabs/packer-plugin-tart/pull/21) This way Packer plugin runs something like `tart run sonoma-base --graphics --vnc-experimental` to have both VNC for the plugin to send commands to and UI window for debugging by a person.

Interesting you should mention that :) One of my use-cases is that I want to run packer provisioning using Tart without any graphics _or audio_. Would you expect the `headless` option in the [Tart packer plugin](https://developer.hashicorp.com/packer/integrations/cirruslabs/tart/latest/components/builder/tart) to then map to both `--no-graphics` and `--no-audio` ?



> In case of audio still seems like a single `--no-audio` mimicking `--no-graphics` is sufficient enough.

Fair enough :)

> I think in most of the cases when `--no-graphics` is used it's used on a host somewhere in a datacenter where either sounds on the host is already disabled or there are no people to bother. 😅 Introducing yet another flag like `--headless` is an option but might be too much for just passing `--no-graphics --no-audio` by an integrator.

For the data center case, I'd agree. But what do you think of a `detach/daemon` option? Having to keep a terminal open (and remembering to not close it to not kill the VM) is a bit annoying :)

**Comment by fkorotkov** [comment]
For the packer plugin we might just add an option to disable audio as well. Might be handy for when we use screen reader 😅 

`detach/daemon/headless` is a theme that was discussed several times on different occasions and we'd agreed in most cases that integrations should handle VM lifecycle and "background" activity. For example, [Cirrus CLI](https://tart.run/integrations/cirrus-cli/) can help you automate your testing. Or if your custom test runner can handle starting and stopping VMs.

We might reconsider developing Tart Desktop application #305 that will sit in your tray. But we need more than just one use case to allocate time for developing and maintaining it.

**Comment by fkorotkov** [comment]
> Why not use the existing `headless` option?

Maybe it can be a part of `headless` but also as a separate option. It's something to think about when making a PR to the Packer plugin. My initial though is to have it configured separately since IMO headless is about no GUI rather than no nothing.  Anyways it's a discussion outside of the scope of this PR.

**Comment by fkorotkov** [comment]
As to merge this PR let's just keep one `--no-audio` option to disable audio device. Does it sound fair?

**Comment by torarnv** [comment]
Yepp, I'll re-push based on that.

**Review by fkorotkov — APPROVED** [review]

**Review by edigaryev — APPROVED** [review]

**Pull request merged**